### DISCUSSION TITLE
How to resolve shadowed error

### DISCUSSION THREAD

**Discussion by roco1234** [open-policy-agent/community#606] [OPA and Rego] [answered]
I get a shadowed error for this function. How should this function be used correctly

>` rego_compile_error: called function ... flatten_arr shadowed`

```
flatten_arr(arr) = flat {
    flat := [item | sublist := arr[_]; item := sublist[_]]
}

allow {
    data := input.UserData
    data != null
    flattened_data := flatten_arr(data)
   #some other checks
}
```

but works with 

```
allow {
    data := input.UserData
    data != null
    flattened_data := [item | sublist := arr[_]; item := sublist[_]]
   #some other checks
}
```

**Comment by srenatus** [comment, marked as answer]
`data` is a special thing in OPA. Can you try to call it `userData` to avoid colliding with that?

**Reply by srenatus** [reply]
[See this playground](https://play.openpolicyagent.org/p/Rpd6mYK5Fx)

**Reply by roco1234** [reply]
ah that fixed the issue thanks! Error message was misleading

**Reply by srenatus** [reply]
Feel free to file an issue for misleading error messages. Please be sure to put enough rego in your issue to let someone reproduce this locally, though 😃
